clients/src/main/java/org/apache/kafka/clients/consumer/internals/CoordinatorRequestManager.java

```diff
@@ -188,6 +190,8 @@ private void onFailedResponse(final long currentTimeMs, final Throwable exceptio
         if (exception == Errors.GROUP_AUTHORIZATION_FAILED.exception()) {
             log.debug("FindCoordinator request failed due to authorization error {}", exception.getMessage());
             KafkaException groupAuthorizationException = GroupAuthorizationException.forGroupId(this.groupId);
+            metadataError.completeExceptionally(groupAuthorizationException);
+            metadataError = metadataError.newIncompleteFuture();
```

Comment:
Although I haven’t implemented it yet, I’ve been considering that this approach would lead to three different ways of handling exceptions in the background thread:
	1.	ErrorEvent
	2.	Completed future
	3.	Exception

It feels a bit inconsistent, but I’ll try this out first. If it works, We can revisit and refactor this issue later.